### DISCUSSION TITLE
shell_init overrides log filters set by modules

### DISCUSSION THREAD

**Discussion by luchnikov** [zephyrproject-rtos/zephyr#55703] [General]
I observed a problem where log_filter_set() called by the driver is overridden on startup. I root-caused it to shell initialization. The temporal sequence is as follows:

1. shell_uart backend initializes at POST_KERNEL level and calls shell_init 
2. shell_init spawns a thread
3. driver initializes at a later POST_KERNEL level and calls log_filter_set() to lower its logging level below compiled-at
4. shell thread finally starts and calls log_filter_set() on every source at the configured serial backend level

Is this intentional? It precludes ability to simultaneously support the following:
(a) module A compiles logging at DBG but wants to come up set to INF
(b) module B compiles logging at DBG and wants to set/log at DBG on startup
If I configure serial backend level to DBG, I violate (a). If I configure serial backend level to INF, I violate (b). In general, shouldn't the backend not override the level on boot if it has already been set previously

**Comment by wilsoncRPE** [comment]
I have also run into this.  What is the workaround to have a module compiled at DBG, but set to INF, when using the shell?

**Comment by paolotreu-lsilastem** [comment]
Same here, is there any update or workaround on this issue? Thanks
